components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/JWTGenerator.java

```diff
@@ -126,75 +130,43 @@
-                        .getValueFromCacheByToken(new AuthorizationGrantCacheKey(accessToken));
-                if (cacheEntry != null) {
-                    customClaimsWithMapping.putAll(cacheEntry.getUserAttributes());
-                }
-            } else if (authCode != null) {
-                AuthorizationGrantCacheEntry cacheEntry = AuthorizationGrantCache.getInstance()
-                        .getValueFromCacheByCode(new AuthorizationGrantCacheKey(authCode));
-                if (cacheEntry != null) {
-                    customClaimsWithMapping.putAll(cacheEntry.getUserAttributes());
-                }
-            } else {
-                customClaimsWithMapping.putAll(validationContext.getUser().getUserAttributes());
-            }
-            String username = validationContext.getValidationInfoDTO().getEndUserName();
-            int tenantId = APIUtil.getTenantId(username);
-
-            customClaims = convertClaimMap(customClaimsWithMapping, username);
-
-            if (isNotEmpty(customClaims)) {
-                if (log.isDebugEnabled()) {
-                    log.debug("The custom claims are retrieved from AuthorizationGrantCache for user : " +
-                            validationContext.getValidationInfoDTO().getEndUserName());
-                }
-            } else {
-                if (log.isDebugEnabled()) {
-                    log.debug("Custom claims are not available in the AuthorizationGrantCache. Hence will be " +
-                            "retrieved from the user store for user : " +
-                            validationContext.getValidationInfoDTO().getEndUserName());
-                }
-            }
-            // If claims are not found in AuthorizationGrantCache, they will be retrieved from the userstore.
+        Map<ClaimMapping, String> customClaimsWithMapping = new HashMap<>();
+        Map<String, String> customClaims;
+        Map<String, Object> properties = new HashMap<String, Object>();
 
-            try {
+        String accessToken = validationContext.getAccessToken();
+        String authCode = validationContext.getAuthorizationCode();
+        if (accessToken != null) {
+            properties.put(APIConstants.KeyManager.ACCESS_TOKEN, accessToken);
+        } else if (authCode != null) {
+            properties.put(APIConstants.KeyManager.AUTH_CODE, authCode);
+        } else {
+            customClaimsWithMapping.putAll(validationContext.getUser().getUserAttributes());
+        }
+        String username = validationContext.getValidationInfoDTO().getEndUserName();
+        int tenantId = APIUtil.getTenantId(username);
 
-                if (tenantId != -1) {
-                    UserStoreManager manager = ServiceReferenceHolder.getInstance().
-                            getRealmService().getTenantUserRealm(tenantId).getUserStoreManager();
+        customClaims = convertClaimMap(customClaimsWithMapping, username);
```

Comment:
this one doesn't need since done in webapp

Reply:
i kept it to convert user attributes coming from validationContext.getUser().getUserAttributes()

Reply:
removed